src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -1157,6 +1156,88 @@ public ImmutableArray<SymbolDisplayPart> ToDisplayParts(SymbolDisplayFormat form
             }
         }
 
+        internal byte? GetNullableContextValue()
+        {
+            var symbol = this;
+            do
+            {
+                var value = symbol.GetLocalNullableContextValue();
+                if (value != null)
+                {
+                    return value;
+                }
+                symbol = symbol.ContainingSymbol;
+            } while (!(symbol is null));
+            return null;
+        }
+
+        internal virtual byte? GetLocalNullableContextValue()
+        {
+            return null;
+        }
+
+        internal void GetCommonNullableValues(CSharpCompilation compilation, ref MostCommonNullableValueBuilder builder)
+        {
+            switch (this.Kind)
+            {
+                case SymbolKind.NamedType:
+                    if (compilation.ShouldEmitNullableAttributes(this))
+                    {
+                        builder.AddValue(this.GetLocalNullableContextValue());
+                    }
+                    break;
+                case SymbolKind.Event:
+                    if (compilation.ShouldEmitNullableAttributes(this))
+                    {
+                        builder.AddValue(((EventSymbol)this).TypeWithAnnotations);
```

Comment:
>builder.AddValue(((EventSymbol)this).TypeWithAnnotations); [](start = 24, length = 58)

Same as for properties, annotations are not emitted on events themselves, only on accessors.